lib/aquifer.api.js

```diff
@@ -74,18 +78,44 @@ class Aquifer {
 
       // Traverse cwd, and find the aquifer project dir. If one does not exist
       // in the current path, the framework is not initialized.
+      this.getProjectDir()
+      .then(() => {
+        if (this.projectDir) {
+          this.initialized = true;
+        }
+        return this.getConfig();
+      })
+      .then(() => {
+        return this.switchVersion();
+      })
+      .then(() => {
+        if (this.initialized) {
+          this.project = new this.api.project(this);
+        }
+        resolve();
+      })
+      .catch((reason) => {
+        reject(reason);
+      })
+    });
+  }
+
+  /**
+   * Find the current Aquifer project root.
+   * @returns {object} promise object.
+   */
+  getProjectDir() {
+    return new Promise((resolve, reject) => {
+      // Traverse cwd, and find the aquifer project dir.
       let dir = this.cwd;
```

Comment:
To clarify I don't mean the results of the traversal but the process of the traversal, we know at this moment all of the directories that we want to check for the existence of aquifer.json. since the only digging we are doing is directory by directory.